pkg/controller/status/status.go

```diff
@@ -216,6 +223,13 @@ func (m *statusManager) AddCronJobs(cjs []types.NamespacedName) {
 	}
 }
 
+// AddCertificateSigningRequests tells the status manager to monitor the health of the given CertificateSigningRequests.
```

Comment:
Would be worth extending the comment here to explain that the "label" argument is the value of the `k8s-app` label on the CSR itself. Alternatively, to make this code more generic, we could pass both the key and value to this function.

As it is, I believe there is some tribal knowledge here, since the current implementation of the status manager code is highly-dependent on the logic of the CSR creation code (the two need to be in-sync on labels).

Reply:
While this is code-wise more succinct I agree that it would be better to stick to the conventional data structures here. I will update my PR accordingly.